scipy/stats/_morestats.py

```diff
@@ -1673,7 +1674,32 @@ def _neg_llf(lmbda, data):
         return -llf
 
     with np.errstate(invalid='ignore'):
-        return optimize.brent(_neg_llf, brack=brack, args=(x,))
+        if not np.all(np.isfinite(x)):
+            raise ValueError('Yeo-Johnson input must be finite.')
+        if brack is not None:
+            return optimize.brent(_neg_llf, brack=brack, args=(x,))
+        x = np.asarray(x)
+        dtype = x.dtype if np.issubdtype(x.dtype, np.floating) else np.float64
+        # Allow values up to 20x the maximum observed value to be safely
+        # transformed without over- or underflow.
+        log1p_max_x = np.log1p(20 * np.max(np.abs(x)))
+        # Use half of floating point's exponent range to allow safe computation
+        # of the variance of the transformed data.
+        log_eps = np.log(np.finfo(dtype).eps)
+        log_tiny_float = (np.log(np.finfo(dtype).tiny) - log_eps) / 2
+        log_max_float = (np.log(np.finfo(dtype).max) + log_eps) / 2
+        # Compute the bounds by approximating the inverse of the Yeo-Johnson
+        # transform on the smallest and largest floating point exponents, given
+        # the largest data we expect to observe. See [1] for further details.
+        # [1] https://github.com/scipy/scipy/pull/18852
+        lb = log_tiny_float / log1p_max_x
+        ub = log_max_float / log1p_max_x
+        # Convert the bounds if the data is negative.
+        if np.any(x < 0):
+            lb, ub = 2 - ub, 2 - lb
```

Comment:
I used `np.any` because it is slightly more defensive in case there are non-finite values such as `NaN` in the data. It could be `np.all` if you prefer that. The min/max suggestion you make is compact, but I'm not sure it's equivalent.

Reply:
I‘m not sure if those reflections formulas apply if a only one single value in x < 0.

Reply:
I improved the bounds so that the cases {all positive, all negative, mixed positive & negative} data are handled separately:
```python
# Compute lb, ub for all positive data
# Reflect the bounds if the data is all negative
if np.all(x < 0):
    lb, ub = 2 - ub, 2 - lb
# Take the most conservative of the positive and negative bounds if the data has mixed signs
elif np.any(x < 0):
    lb, ub = max(2 - ub, lb), min(2 - lb, ub)
```